ksqldb-serde/src/main/java/io/confluent/ksql/serde/avro/KsqlAvroSerdeFactory.java

```diff
@@ -140,6 +140,9 @@ private static AvroConverter getAvroConverter(
 
     avroConfig.put(AvroDataConfig.CONNECT_META_DATA_CONFIG, false);
 
+    avroConfig.put(KsqlConfig.KSQL_VALUE_SUBJECT_NAME_STRATEGY,
+            ksqlConfig.getString(KsqlConfig.KSQL_VALUE_SUBJECT_NAME_STRATEGY));
```

Comment:
I finally had a chance to look into the contents of the PR. 

I'm worried that it's not as simple as this - we have quite a bit of code that relies on the `TopicNameStrategy`. For example, see usages of `KsqlConstants#getSRSubject`. 
- One such use case is that we want to subject to exist before any data exists in the topic (see https://github.com/confluentinc/ksql/issues/4219 and the crazy chain of issues that arose around that).
- Another is when we clean up a ksqlDB cluster, we look for all the subjects that we created and clean them up as well (you can see the code in `ClusterTerminator`) which also hardcodes the dependency on the TopicNameStrategy.

If we do manage to make all of the above generic (which may involve non-trivial amounts of work), I think we also need to explicitly allow-list the subject strategies that are supported. Some strategies (e.g. `RecordNameStrategy` don't play very well with ksqlDB's assumption that each topic only has one schema).

Let me know if this makes sense, but I think there's a non-trivial amount of work if we want this to work natively end-to-end (it probably even warrants a KLIP)

Reply:
Yeeees. Basically, my patch is work (create tables and streams, read from topics), but yes, it is failed e.g. for statements create a table from external connector (Postgres in my case). I also debug a few hours of usages `KsqlConstants#getSRSubject`. And thank you for the explanation.

Meantime `KsqlConstants#getSRSubject` static. From the first point of view, we should not put there some logic like read configs (is it possible in such context?). On another side, if is public, it is API, so it is a good idea to deprecate it and support the strategy.

What is looks more reasonable to me is move such logic into `io.confluent.ksql.schema.ksql.inference.SchemaRegistryTopicSchemaSupplier`. Read configuration and in instantiation also create a configured strategy. What do you think?

Reply:
And then we may just refactor and delete `KsqlConstants#getSRSubject`

What you speak about possible `RecordNameStrategy` if it can't be used, we may just document it and give correct validation message.

Reply:
Could you please say or provide an example of what will be a correct way to get access to a parsed instance of `KsqlConfig` from within `SchemaRegistryTopicSchemaSupplier`?

Reply:
@agavra, could you please look at https://github.com/confluentinc/ksql/pull/6733/commits/1d4fe5f98c735bb13c1acc1c1f4063caa1f7c0fb commit?

I do not found a better way to inject configurated strategy into KsqlConstants and leave `static getSRSubject(...)` for backward compatibility.

Even if I introduce some DI mechanism and provide the bean like `subjectNameStrategy`, if I understand correctly we need to save public `getSRSubject`? Right?

Reply:
Thanks @Hubbitus - I took another look. I think there's a good chunk of design work to be done before we can start thinking about the best approach (and it needs to be holistic, not just on the write-path). If we try to stick something in now, we'll be stuck with supporting it for the foreseeable future (and we have to make sure we don't break anything either). 

Is it blocking you to have ksql register things under the wrong subject name? You can always copy the schema across to be what you need it to be downstream.

Reply:
> I think there's a good chunk of design work to be done before we can start thinking about the best approach (and it needs to be holistic, not just on the write-path).
It is not write-path related! Furthermore, my first attempt which you comment, was the opposite, to support read data from topics as ksqldb streams and tables!

> If we try to stick something in now, we'll be stuck with supporting it for the foreseeable future (and we have to make sure we don't break anything either).
I agree I can miss something. It is my first contribution attempt. Maybe I need to write or extend some tests? Integration I think more interesting for that.

> Is it blocking you to have ksql register things under the wrong subject name? You can always copy the schema across to be what you need it to be downstream.
As I stated before ksqldb even can't read our topics! When I start to try to add source connectors, I was forced then configure schemes registration too. I had tried copy schemes under different names, on sandbox but it also near impossible when there several versions (or I do not know how).
So, really it is fully blocking us use ksqldb in our environment. Or use the patched version (that also is not a good alternative in the long-term perspective).

Reply:
@agavra, do you reject merging that pull request, or I can do something to move forward with it? I've ready to extend tests or adjust implementation.

Reply:
Hello @Hubbitus - I don't think our team has enough bandwidth to help get this change across the line (as I mentioned, I think there's still a lot of work to be done). So I guess I'm "rejecting" it (though it's not because I don't think this feature is valuable, but I think it requires a lot more design from people familiar with all the implications).

Reply:
Maybe then you will be so kind as to plan it into some roadmap (milestone)?
And again, please say if I can doing something to help it moving forward.

Reply:
cc @colinhicks for roadmapping purposes